### PR TITLE
Configure should be called in the current thread and not in a UI thread

### PR DESCRIPTION
**Pull request by vegaro**
We were calling configure in the UI thread (main)

This is the thread where Cordova Java code runs. Note is in the `main` pool but it's not the main thread. 

Screen Shot 2022-09-29 at 5 00 26 PM

We had `ExecutionThread.UI` as the thread configure should be running on. So this code in the annotation plugin was running on that JavaBridge thread:

```
Object[] mArgs = getMethodArgs(args, callbackContext);
Runnable runnable = this.createRunnable(mArgs, caller, callbackContext);
if (executionThread == ExecutionThread.WORKER) {
  cordova.getThreadPool().execute(runnable);
} else if (executionThread == ExecutionThread.UI) {
  cordova.getActivity().runOnUiThread(runnable);
} else {
  runnable.run();
}
```

And calling `configure` in our plugin in the `main` thread in the `main` pool. Setting a breakpoint inside `configure` in phc shows this thread:

Screen Shot 2022-09-29 at 5 26 56 PM

This means making these Javascript calls:

```
    Purchases.configure("api_key");
    Purchases.isAnonymous(
      isAnonymous => {
        console.log("ISANONYMOUS " + isAnonymous);
        }
    );
```

Would trigger the following exception:

```
Uncaught exception at @isAnonymous
kotlin.UninitializedPropertyAccessException: There is no singleton instance. Make sure you configure Purchases before trying to get the default instance. More info here: https://errors.rev.cat/configuring-sdk
  at com.revenuecat.purchases.Purchases$Companion.getSharedInstance(Purchases.kt:1985)
  at com.revenuecat.purchases.hybridcommon.CommonKt.isAnonymous(common.kt:245)
  at com.revenuecat.purchases.PurchasesPlugin.isAnonymous(PurchasesPlugin.java:176)
  at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	...
```

The reason for the exception is that, since we are changing threads (from `JavaBridge` to `main`), the configure Javascript call returns before completing the code in `configure` and `isAnonymous` gets called (in `JavaBridge`) and it executes before the singleton instance is set.

Changing `configure` to execute in `ExecutionThread.MAIN` fixes it, since it will block that thread and no other plugin function will be called until it completes.

This is another approach to fix the same as https://github.com/RevenueCat/cordova-plugin-purchases/pull/187

[CSDK-481]

[CSDK-481]: https://revenuecats.atlassian.net/browse/CSDK-481?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/android/PurchasesPlugin.java [hunk 1]
```diff
@@ -35,7 +35,7 @@ public class PurchasesPlugin extends AnnotatedCordovaPlugin {
     public static final String PLATFORM_NAME = "cordova";
     public static final String PLUGIN_VERSION = "3.0.0-rc.8";
 
-    @PluginAction(thread = ExecutionThread.UI, actionName = "configure", isAutofinish = false)
+    @PluginAction(thread = ExecutionThread.MAIN, actionName = "configure", isAutofinish = false)
```
Comment (aboedo) [inline src/android/PurchasesPlugin.java:38]: awesome! but let's definitely leave a comment here explaining why this is important

**Review by aboedo — COMMENTED** [review]
can we also update the sample app so it confidently relies on this behavior instead of waiting for the delegate call?

**Comment by aboedo** [comment]
@vegaro we didn't but not because of solid design, it was just a byproduct of the fact that on iOS we have to do some thread switching to figure out whether the app is running in the foreground or background

**Review by aboedo — APPROVED** [review]
it'll be fun to rebase [this one](https://github.com/RevenueCat/cordova-plugin-purchases/pull/186/) off of the changes 😂 
Thanks for taking care of it!!

**Review by tonidero — APPROVED** [review]
Ah interesting knowledge. Thanks!

**Pull request merged**